contracts/mainnet/connectors/morpho-compound/main.sol

```diff
@@ -0,0 +1,440 @@
+	{
+		uint256 _amt = getUint(_getId, _amount);
+
+		bool _isETH = _tokenAddress == ethAddr;
+
+		TokenInterface _tokenContract = _isETH
+			? TokenInterface(wethAddr)
+			: TokenInterface(_tokenAddress);
+
+		if (_amt == uint256(-1)) {
+			_amt = _isETH
+				? address(this).balance
+				: _tokenContract.balanceOf(address(this));
+		}
+
+		if (_isETH) convertEthToWeth(_isETH, _tokenContract, _amt);
+
+		approve(_tokenContract, address(morphoCompound), _amt);
+		morphoCompound.supply(_poolTokenAddress, _onBehalf, _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogDeposit(address,address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_onBehalf,
+			_amt,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Borrow ETH/ERC20_Token.
+	 * @notice Borrow a token from Morpho Compound.
+	 * @param _tokenAddress The address of underlying token to borrow.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of cToken to borrow.(For ETH: cWETH address)
+	 * @param _amount The amount of the token (in underlying) to borrow.
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens borrowed.
+	 */
+	function borrow(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		uint256 _amount,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+
+		bool _isETH = _tokenAddress == ethAddr;
+
+		morphoCompound.borrow(_poolTokenAddress, _amt);
+
+		if (_isETH) convertWethToEth(_isETH, TokenInterface(wethAddr), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogBorrow(address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_amt,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Borrow ETH/ERC20_Token.
+	 * @notice Borrow a token from Morpho Compound with max gas.
+	 * @param _tokenAddress The address of underlying token to borrow.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of cToken to borrow.(For ETH: cWETH address)
+	 * @param _amount The amount of the token (in underlying) to borrow.
+	 * @param _maxGasForMatching The maximum amount of gas to consume within a matching engine loop.
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens borrowed.
+	 */
+	function borrowWithMaxGas(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		uint256 _amount,
+		uint256 _maxGasForMatching,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+
+		bool _isETH = _tokenAddress == ethAddr;
+
+		morphoCompound.borrow(_poolTokenAddress, _amt, _maxGasForMatching);
+
+		if (_isETH) convertWethToEth(_isETH, TokenInterface(wethAddr), _amt);
```

Comment:
```suggestion
		convertWethToEth(_isETH, TokenInterface(wethAddr), _amt);
```
`_isETH` is checked inside `convertWethToEth` so it's currently checked twice without intermediary code execution